### PR TITLE
Update PyTorch getting started guide to point to SHARK

### PR DESCRIPTION
**Pull request by dellis23**
iree-torch is no longer being actively maintained, so we don't want to link to a library that will drift and break.

### CODE REVIEW

**Comment by dellis23** [comment]
Fixes #12034

File: docs/website/docs/index.md [hunk 1]
```diff
@@ -23,7 +23,6 @@ IREE supports importing from a variety of ML frameworks:
 - [x] TensorFlow
 - [x] TensorFlow Lite
 - [x] JAX
-- [x] PyTorch
 - [ ] ONNX (hoped for)
```
Comment (ScottTodd) [inline docs/website/docs/index.md:26]: I'd like to talk through this more before dropping PyTorch from this list entirely (that seems like a much larger topic than "quietly remove it from the docs"). As mentioned on https://github.com/iree-org/iree/issues/12034#issuecomment-1423042586, can we start by pointing users at https://github.com/nod-ai/SHARK or somewhere other than iree-torch?

The JAX page can be an example of a "under construction" page: https://iree-org.github.io/iree/getting-started/jax/

We really need to stand behind commitments like framework support though. I'm pretty concerned that these pages and integrations are slipping.

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/website/docs/index.md (hunk 1 above)
Comment (dellis23) [reply docs/website/docs/index.md:26]: I can update to link to SHARK, but I don't know what we'd put on an under construction page for PyTorch.  The JAX page says it's being actively worked on; in this case, however, it isn't.

Comment (dellis23) [reply docs/website/docs/index.md]: Updated the PR based on our chat offline (used the language you suggested)

**Review by silvasean — COMMENTED** [review]
Is there a more specific thing we can link to than just SHARK? Like does SHARK have a "getting started for PyTorch" docs page? (maybe we could request that?)

**Comment by ScottTodd** [comment]
> Is there a more specific thing we can link to than just SHARK? Like does SHARK have a "getting started for PyTorch" docs page? (maybe we could request that?)

@powderluv (or someone else who works on SHARK) WDYT? I also get the sense that this (general status of PyTorch->IREE support, how much to point IREE docs at SHARK and other community projects) calls for a bit more than a discussion via pull request review comments, just not totally sure who to involve in the discussion.

**Comment by powderluv** [comment]
We can point to [SHARK](https://github.com/nod-ai/SHARK) main page like in the PR, and also point to the [setup](https://github.com/nod-ai/SHARK#setup-a-new-pip-virtual-environment) and an API example of using [torch](https://github.com/nod-ai/SHARK#shark-inference-api).  And if you want we can add a special landing page but I think it can be captured in the main page. 

@monorimet in case we need any changes to our landing pages.

File: docs/website/docs/getting-started/pytorch.md [hunk 2]
```diff
@@ -1,124 +1,5 @@
...
-| -- | -- |
-Inference on BERT | [![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/iree-org/iree-torch/blob/main/examples/bert.ipynb)
-
-| Example scripts |
-| -- |
-| [Basic Inference and Training Example](https://github.com/iree-org/iree-torch/blob/main/examples/regression.py) |
-| [Native On-device Training Example](https://github.com/iree-org/iree-torch/tree/main/examples/native_training) |
+PyTorch support is community driven. To import programs from PyTorch and use
+them with IREE, see projects like [SHARK](https://github.com/nod-ai/SHARK) from
+our partners at nod.ai.
```
Comment (ScottTodd) [inline docs/website/docs/getting-started/pytorch.md:5]: Maybe we should also mention https://github.com/llvm/torch-mlir ? 

Is https://github.com/openxla/iree/tree/main/compiler/src/iree/compiler/InputConversion/TMTensor still relevant?

I haven't been following this side of the project very closely, but it seems like enough pieces exist in MLIR/Torch/IREE to compile and run programs, and SHARK provides more docs/APIs/samples. Still seems good to mention the community aspect and point to examples in SHARK (or other projects) that show e2e workflows, but I think we could elaborate more here.

Comment (powderluv) [reply docs/website/docs/getting-started/pytorch.md:5]: TM Tensor is very important for complex models.

**Comment by ScottTodd** [comment]
Should we close this in favor of https://github.com/openxla/iree/pull/13574?

**Comment by dellis23** [comment]
SGTM

**Pull request closed without merging**